### ISSUE TITLE
Time is incorrect in leaves

### ISSUE THREAD

**Issue by meridasoft** [odoo/odoo#35363]
Impacted versions: 12

Steps to reproduce: The hours to select in leave is inverted 12 AM is 12 PM and 12 PM is 12 AM

Current behavior: When select 12 AM or 12PM the calcultarion of hours is incorrect because is inverted

Expected behavior: Correct calculation

Video/Screenshot link (optional):

**Comment by pedrobaeza** [comment]
Is this happening also on latest code version? You can try on runbot.

@sswapnesh I remember you reported something similar. Can it be?

**Comment by sswapnesh** [comment]
I don't have references of the Wrong Calculations but https://github.com/odoo/odoo/pull/29803 refers to the AM/PM inconsistency.

**Comment by pedrobaeza** [comment]
Yeah, that was, but it's not the same thing, so tagging for later check by Odoo team.

**Comment by sswapnesh** [comment]
@meridasoft  Can you please some add screenshot/Additional details of wrong calculations ?

**Comment by meridasoft** [comment]
for wrong calculation, i have problems with employees with diferent time zone, when select a leves of days show parcial day but it should be a integer day.

![image](https://user-images.githubusercontent.com/44324313/62310473-7d059280-b43e-11e9-8b01-07399cc46c8d.png)

**Comment by brian10048** [comment]
I am noticing the same issue.  It seems to be related to the way `number_of_days` is calculated.

**Take all day off**
Expected: 1 day
Result: Half a day
![image](https://user-images.githubusercontent.com/25710115/67866731-c5952080-faff-11e9-9e83-f07c8bbeca4a.png)

**Take half day in the morning**
Expected: 0.5 day
Result: 0.38 days
![image](https://user-images.githubusercontent.com/25710115/67867196-6a176280-fb00-11e9-84fc-6deb794acd4d.png)

**Take half day in the afternoon**
Expected: 0.5 day
Result: 0 days
![image](https://user-images.githubusercontent.com/25710115/67867230-7996ab80-fb00-11e9-81af-48b384a55103.png)

**Tree view showing actual times of leave**
08:00:00 to 17:00:00 is a full day of work, but calculated as a half day
![image](https://user-images.githubusercontent.com/25710115/67866811-e52c4900-faff-11e9-9dbd-b340e74b9b6c.png)


**The Resource calendar being used**
![image](https://user-images.githubusercontent.com/25710115/67866934-0b51e900-fb00-11e9-8041-2e4a28899633.png)

**Comment by brian10048** [comment]
**Correction to above**

@meridasoft - My issue was resolved by matching the employee's timezone with the resource calendar timezone.  It definitely seems related to time zones, but it may just be a configuration issue

**Comment by gbeuning** [comment]
Note that there might be old translations around which are no longer correct after the reversal of 12:00 AM and 12:00 PM strings in #29803. This may lead to the apparent inversion as initially reported in this issue.
